src/modules/ibc.ts

```diff
@@ -155,14 +164,116 @@ export class IBCModule extends BaseModule {
     return chainInfoMap;
   }
 
-  async getChainInfo(chainName: string, chainId: string): Promise<ChainInfo | undefined> {
+  async getChainInfo(chainName: string): Promise<ChainInfo | undefined> {
     if (!chainName) return undefined
-  
-    try {
-      const chainInfoResponse = await fetch(`https://raw.githubusercontent.com/chainapsis/keplr-chain-registry/master/cosmos/${chainName}.json`);
-      const chainInfoJson = await chainInfoResponse.json();
-      return chainInfoJson as ChainInfo;
-    } catch (err) {
+    const chainInfoResponse = await fetch(`https://raw.githubusercontent.com/chainapsis/keplr-chain-registry/master/cosmos/${chainName}.json`)
+    if (!chainInfoResponse.ok) {
+      if (chainInfoResponse.status === 404){
+        return undefined;
+      } 
+    }
+    const chainInfoJson = await chainInfoResponse.json();
+    return chainInfoJson as ChainInfo;
+  }
+
+  async getAssembledChainInfo(chainsData: ChainRegistryItem[], chainId: string, bridgeChainName: string): Promise<ChainInfo | undefined> {
+    const selectedChainData = chainsData.find((chainData: ChainRegistryItem) => {
+      return chainData.chain_name.includes(bridgeChainName.toLowerCase()) || chainData.chain_id.includes(bridgeChainName.toLowerCase());
+    });
+
+    if (selectedChainData) {
+      try {
+        const chainInfoResponse = await fetch(`https://raw.githubusercontent.com/cosmos/chain-registry/master/${selectedChainData.chain_name}/chain.json`);
+        const chainInfoJson = await chainInfoResponse.json();
+        const chainAssetListResponse = await fetch(`https://raw.githubusercontent.com/cosmos/chain-registry/master/${selectedChainData.chain_name}/assetlist.json`);
+        const assetListJson = await chainAssetListResponse.json();
+        const features: string[] = [];
+      if (selectedChainData.cosmwasm_enabled) {
+        features.push("cosmwasm");
+      }
+
+      // Extract the staking currency denomination from chainInfoJson (min. denom)
+      const stakingCurrencyDenom = chainInfoJson['staking']['staking_tokens'][0]['denom'];
```

Comment:
1. Actually, you don't need to use square brackets. Can use dot notation directly.
2. When you are using number as an index (i.e. for array), make sure to use optional chaining and have a fallback value.
i.e.
```typescript
const stakingCurrencyDenom = chainInfoJson.staking.staking_tokens[0]?.denom ?? '';
```

Reply:
I've made the change, thank you!